scripts/inventory/consul_io.py

```diff
@@ -312,10 +312,10 @@ def load_data_for_node(self, node, datacenter):
         '''loads the data for a single node adding it to various groups based on
         metadata retrieved from the kv store and service availability'''
 
-        if self.config.suffixes == 'true':
-            index, node_data = self.consul_api.catalog.node(node, dc=datacenter)
-        else:
+        if self.config.bulk_load == 'true':
```

Comment:
I'm not sure what this change is supposed to be doing. Could you please explain why you are changing this to use `bulk_load` instead of `suffixes`?

Reply:
Hi @felixfontein,
Thanks for the review
From my understanding, `consul_get_node_inmemory` should only be called when `bulk_load` was set to `true`
and the current code would only fetch the node data from Consul when `suffixes` was set to `true`
(take `load_data_for_datacenter` as a example)

So the change would make the code decide whether  to  fetch data from Consul by the value of `bulk_load`
but not `suffixes`

Reply:
This change will not solve the issue entirely because the script fails to properly handle the bulk_load configuration option. The proper fix would be to make sure data is actually bulk-loaded if required by the user. But as far as this part of the code goes, this change is correct.